01_roman_numerals.py

```diff
@@ -0,0 +1,191 @@
+class RomanNumerals:
+  """
+  I - 1
+  V - 5
+  X - 10
+  L - 50
+  C - 100
+  D - 500
+  M - 1000
+  """
+
+  def roman_to_number(self, string):
+    """
+    Converts a string to roman numerals.
+    Assumes that everything in the string is a valid roman numeral
+    Subtraction is not present. ie: IX = 9 does not exist. 9 would be XIIII
+
+    Analysis
+    Time - o(n)
+    * have to iterate through the entire string once. n is the length of the string
+    Space - o(1)
+    * size of lookup table is constant since there is a fixed set of roman numerals
+    * rest are variables that are integers
+    """
+    lookup = {"I": 1, "V": 5, "X": 10, "L": 50, "C": 100, "D": 500, "M": 1000}
+
+    acc = 0
+
+    for c in string:
+      acc += lookup[c]
+
+    return acc
+
+  def roman_to_number_2(self, string):
+    """
+      Converts a string to roman numerals.
+      Assumes that everything in the string is a valid roman numeral that follows the rules below.
+      Subtraction is present. ie: IX = -1+10 = 9
+      Numbers go bigger to smaller unless subtraction is present.
+      Not all pairs are allowed.
+      * I can precede V or X
+      * X can precede L or C
+      * C can precede D or M
+
+      Analysis
+      Time - o(n) where n is the length of the string
+      * no matter what, we have to iterate through each character in the
+      entire string once
+      Space - o(1)
+      * lookup table is a constant
+      * temporary variables store either fixed sized strings or integers
+    """
+    lookup = {"I": 1, "V": 5, "X": 10, "L": 50, "C": 100, "D": 500, "M": 1000}
+
+    acc = 0
+    idx = 0
+    str_length = len(string)
+
+    while idx < str_length:
```

Comment:
Great!  I think this handles the edge cases nicely.

One way you might split things up even more is to introduce a separate function like "should_subtract" or "is_subtractive_pair".